cpp/src/arrow/array/builder_base.h

```diff
@@ -140,6 +140,8 @@ class ARROW_EXPORT ArrayBuilder {
   /// Set the next length bits to not null (i.e. valid).
   Status SetNotNull(int64_t length);
 
+  void ResetBase();
```

Comment:
It would be nice to provide some documentation for how/when this method should be used.

Reply:
Agreed. Ideally we should be able to avoid having this new method?